akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -388,6 +396,19 @@ final class ExplicitlyAskableActorRef(val actorRef: ActorRef) extends AnyVal {
   def ?(message: ActorRef => Any)(implicit timeout: Timeout, sender: ActorRef = Actor.noSender): Future[Any] =
     internalAsk(message, timeout, sender)
 
+  def askWithStatus(
+      message: ActorRef => Any)(implicit timeout: Timeout, sender: ActorRef = Actor.noSender): Future[Any] =
+    internalAskWithStatus(message, timeout, sender)
+
+  /**
+   * INTERNAL API
+   */
+  private[pattern] def internalAskWithStatus(
+      messageFactory: ActorRef => Any,
+      timeout: Timeout,
+      sender: ActorRef): Future[Any] =
+    StatusReply.flattenStatusFuture[Any](internalAsk(messageFactory, timeout, sender).mapTo[StatusReply[Any]])
```

Comment:
I'm not sure if we should add `askWithStatus` to the classic pattern. At the end, the only thing we need is to flatten the status. 

We could eventually do it inside the flow implementation. 

@johanandren, any reason the classic `AskSupport` doesn't have a `askWithStatus`? Was that an oversight or we then decided not to add it?

Reply:
There is ask with status in classic already: https://github.com/akka/akka/blob/master/akka-actor/src/main/scala/akka/pattern/AskSupport.scala#L96
and https://github.com/akka/akka/blob/master/akka-actor/src/main/scala/akka/pattern/Patterns.scala#L103

Reply:
I was not sure how to add askWithStatus only to typed streams (without adding askWithStatus to akka-actor/src/main/scala/akka/pattern/AskSupport.scala )  .
I'll investigate further.

Reply:
I see, the one being added here is with explicit sender. I missed that detail.

Reply:
no extra askWithStatus on classic.
added askWithStatus.md but I left a TODO on the Signature since I don't know how to generate it